#### Q1

In [3]:
'''
   Generate a binomial distribution, tested 10 times, given the number of trials(n) and probability(p) of each trial.
   The input will contain seed, n and p in the same order.
   The output should contain a numpy array with 10 numbers representing the required binomial distribution.
   Explanation: 
   seed (integer) → Used to initialize NumPy’s random number generator for reproducibility.
   n (integer)    → Number of trials in each binomial experiment.
   p (float)      → Probability of success in each trial.
 
'''
import numpy as np 
seed=int(input())
n=int(input())
p=float(input())
#write your code here
# Set the seed for reproducibility
np.random.seed(seed)

# Generate binomial distribution (10 samples)
s = np.random.binomial(n, p, 10) 
print(s)

 2
 8
 .3


[2 0 2 2 2 2 1 3 2 2]


#### Q2

In [4]:
''' 
    A card stack contains white and black cards. Two cards are drawn randomly without replacement. 
    The probability of selecting a white  and then a black card is x. 
    The probability of selecting a white card in the first draw is y. 
    You have to find the probability of drawing a black card, given that the first card drawn was white.
    The input will contain two lines with x and y respectively.  
    The output should be displayed as a float(no need to round it off).
    Sample Input:
    0.2
    0.5
    Sample Output:
    0.4
'''
# Take inputs
x = float(input())  # Probability of drawing white then black
y = float(input())  # Probability of drawing white first

# Compute conditional probability
result = x / y

# Print the result
print(result)


 .5
 .2


2.5


#### Q3

In [ ]:
'''
    N people sit around a circular table. You have to find the probability that two particular people won't be sitting together.
    The input will have the number N and the output should have the probability printed as a float type number rounded off to four decimal places.
    Hint: This problem follows the logic of circular permutation which you can read about here.
    Please note that if the answer is 3.6, it is to be reported as 3.6 and not 3.6000

    Sample Input:
    15
    Sample Output:
    0.8571
'''
import math

# Take input
N = int(input())

# Compute probability
if N > 2:
    probability_not_together = 1 - (2 / (N - 1))
else:
    probability_not_together = 0.0  # If only 2 people, they must sit together

# Print result rounded to 4 decimal places
print(round(probability_not_together, 4))


#### Q4
![Sample Image](Formula_1.png)

In [6]:
import numpy as np

# Given population data
population = np.array([8, 9, 7, 10, 6, 8, 9, 7, 10, 6])

# Sample data (randomly chosen)
sample = np.array([8, 9, 7, 10, 6])

# Population Mean
population_mean = np.mean(population)

# Sample Mean
sample_mean = np.mean(sample)

# Population Standard Deviation (using N, so ddof=0)
population_sd = np.std(population, ddof=0)

# Sample Standard Deviation (using n-1, so ddof=1)
sample_sd = np.std(sample, ddof=1)

# Standard Error (SE)
sample_size = len(sample)
standard_error = sample_sd / np.sqrt(sample_size)

# Print results
print(f"Population Mean (μ): {population_mean}")
print(f"Sample Mean (X̄): {sample_mean}")
print(f"Population SD: {population_sd}")
print(f"Sample SD: {sample_sd}")
print(f"Standard Error (SE): {standard_error}")


Population Mean (μ): 8.0
Sample Mean (X̄): 8.0
Population SD: 1.4142135623730951
Sample SD: 1.5811388300841898
Standard Error (SE): 0.7071067811865476


Interpretation
1️⃣ Population Mean (𝜇μ) = 8 hours → The true average working hours for all 10 employees.
2️⃣ Sample Mean (𝑋ˉ) = 8 hours → The estimated average based on the sample.
3️⃣ Population SD (𝑆𝐷pop) = 1.41 → Shows variability in the full dataset.
4️⃣ Sample SD (𝑆𝐷sample) = 1.58 → Measures variation within the sample.
5️⃣ Standard Error (SE) = 0.71 → Indicates how much sample means typically deviate from the true population mean.